tests/integration/test_logcollector/test_configuration/test_basic_configuration_target.py

```diff
@@ -0,0 +1,110 @@
+    prefix = AGENT_DETECTOR_PREFIX
+else:
+    prefix = LOG_COLLECTOR_DETECTOR_PREFIX
+
+parameters = [
+    {'SOCKET_NAME': 'custom_socket', 'SOCKET_PATH': '/var/log/messages', 'LOCATION': "/tmp/testing.log",
+     'LOG_FORMAT': 'syslog', 'TARGET': 'custom_socket'},
+    {'SOCKET_NAME': 'custom_socket', 'SOCKET_PATH': '/var/log/messages', 'LOCATION': "/tmp/testing.log",
+     'LOG_FORMAT': 'json', 'TARGET': 'custom_socket'},
+    {'SOCKET_NAME': 'custom_socket2', 'SOCKET_PATH': '/var/log/messages', 'LOCATION': "/tmp/testing.log",
+     'LOG_FORMAT': 'json', 'TARGET': 'custom_socket'},
+]
+metadata = [
+    {'socket_name': 'custom_socket', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'syslog', 'target': 'custom_socket', 'valid_value': True},
+    {'socket_name': 'custom_socket', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'json', 'target': 'custom_socket', 'valid_value': True},
+    {'socket_name': 'custom_socket2', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'json', 'target': 'custom_socket', 'valid_value': False},
+]
+
+configurations = load_wazuh_configurations(configurations_path, __name__,
+                                           params=parameters,
+                                           metadata=metadata)
+configuration_ids = [f"{x['LOG_FORMAT'], x['TARGET'], x['SOCKET_NAME'], x['LOCATION'], x['SOCKET_PATH']}"
+                     for x in parameters]
+
+
+def check_configuration_target_valid(cfg):
+    """Check if the Wazuh module run correctly and that it uses the designated socket.
+
+    Ensure logcollector is running with the specified configuration, analyzing the designated socket and
+    , in case of the Wazuh server, check if the API answer for localfile configuration block coincides
+    the selected configuration.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If the socket target callback is not generated.
+        AssertError: In case of a server instance, the API response is different that the real configuration.
+    """
+    log_callback = logcollector.callback_socket_target(cfg['location'], cfg['target'], prefix=prefix)
+    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_TARGET_SOCKET)
+
+    if wazuh_component == 'wazuh-manager':
+        real_configuration = dict((key, cfg[key]) for key in ('location', 'target', 'log_format'))
+        api.wait_until_api_ready()
+        api.compare_config_api_response([real_configuration], 'localfile')
+
+
+def check_configuration_target_invalid(cfg):
+    """Check if the Wazuh fails because a invalid target configuration value.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If error callback are not generated.
+    """
```

Comment:
```suggestion
    """Check if Wazuh fails because of an invalid target configuration value.

    Args:
        cfg (dict): Dictionary with the localfile configuration.

    Raises:
        TimeoutError: If the error callbacks are not generated.
    """
```